### DISCUSSION TITLE
34th open maintainers call

### DISCUSSION THREAD

**Discussion by mxinden** [libp2p/rust-libp2p#5017] [General]
Hi everyone,

The rust-libp2p maintainers will meet tomorrow (Tuesday) 11:00 UTC. See [libp2p public events calendar](https://calendar.google.com/calendar/embed?src=libp2p.io_0q9682i3te7eanhe9q7ae1c58g%40group.calendar.google.com).

https://meet.jit.si/rust-libp2p-open-maintainers-call

Everyone is welcome to join, listen along, propose topics and join the discussions.

Please post below in case you already want to add an agenda item or list it in [our notes document](https://hackmd.io/xoHKlFuER0OVj1DGDuFROw).

See you all tomorrow!

**Comment by dgarus** [comment]
Hi!
**webtransport: use deterministic TLS certificates**

I tried to find out how to generate a self-signed certificate deterministically for `webtransport` integration.

There was a Max's suggestion to use an approach similar to implemented in [go-libp2p](https://github.com/libp2p/rust-libp2p/pull/4874#discussion_r1402683019). But I faced a problem:
`EcdsaKeyPair::generate_pkcs8(...)`  take an impl of the `rand::SecureRandom` trait, but it can't be implemented outside the `rand` crate. It made intentionally that follows from his module name `sealed`.
So I don't see any way to use `ring` crate and be able to generate a key deterministically. 
Maybe you see.

**Reply by thomaseizinger** [reply]
I had already started a discussion with `rcgen` for this, see https://github.com/rustls/rcgen/issues/173. No solution yet unfortunately.

**Reply by dgarus** [reply]
As a temporary solution, we could use for `webtransport` cert generation another library that more fit our requirements. 
As life shows, temporary solutions are the most permanent :D

**Reply by thomaseizinger** [reply]
> As a temporary solution, we could use for `webtransport` cert generation another library that more fit our requirements.

Yes I think that is what we might have to end up doing!
